spec/models/shared_examples/shared_examples_for_loadable.rb

```diff
@@ -43,6 +43,12 @@
         expect(results.num_inserts).to eq(1)
         expect(results.ids.length).to eq(1)
       end
+
+      it 'does not delete previous record if upload is invalid' do
```

Comment:
This is not a failing test before the changes.  Using `byebug` I noticed that `described_class.count` is 5 , which is not what `initial_result_count` is set to.  The `change` matcher expects the change to happen inside the block given to `expect`, and by assigning a value to `results` means no action is being taken and this expect will never see a change.

Also, `transaction` doesn't rollback unless there's an exception, so this test should probably use something like `allow(described_class).to receive(:load_csv_file).and_raise("error in csv")` before `load` is called so that there's an exception.